unified-codes/apps/data-service/src/app/resolvers.js

```diff
@@ -29,22 +30,25 @@ const queries = {
 
 export const resolvers = {
   Query: {
-    entity: async (_source, _args, { dataSources }) => {
+    entity: async (_source, _args, context) => {
+      const { dataSources } = context;
       const { code } = _args;
       const query = queries.entity(code);
       const response = await dataSources.dgraph.postQuery(query);
       const [entity] = response.data.query;
       return entity;
     },
-    entities: async (_source, _args, { dataSources }) => {
+    entities: async (_source, _args, context) => {
+      const { dataSources } = context;
       const { type = 'medicinal_product' } = _args?.filter ?? {};
       const query = queries.entities(type);
       const response = await dataSources.dgraph.postQuery(query);
       return response.data.query;
     },
     user: async (_parent, _args, context) => {
-      const { user } = context;
+      const { user, authenticationService, authorisationService } = context;
```

Comment:
Idea here is that the two auth services basically represent the source of truth on everyone who exists and everything that they can do, so should provide the resolver with access to everything it needs to decide what can be read/mutated.